### PR TITLE
[BUGFIX] Stop relying on transitive dependencies

### PR DESCRIPTION
**Pull request by oliverklee**
If `psr/http-message` is not installed, we now fail hard with a helpful exception message.

### CODE REVIEW

**Comment by sascha-egerer** [comment]
Maybe we should better not use a "use" statement and do a "class exists" on the FQN? Because we don't really rely on this package - it's just required for the analysis where this package is used.

**Comment by sabbelasichon** [comment]
My first gut feeling: I think, we don´t need to make this explicit. I would rather like if the it fails hard, when the dependency is removed under the hood.

**Comment by oliverklee** [comment]
I have updated the PR accordingly.

Note: `use`ing classes/interfaces that might or might not exist is not a problem as this only is a name aliasing, but does not (try to) load the classes/interfaces yet.

File: src/Type/RequestDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -31,7 +32,7 @@ public function __construct(array $requestGetAttributeMapping, TypeStringResolve
 
 	public function getClass(): string
 	{
-		return \Psr\Http\Message\ServerRequestInterface::class;
+		return interface_exists(ServerRequestInterface::class) ? ServerRequestInterface::class : '';
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply src/Type/RequestDynamicReturnTypeExtension.php:35]: We could also throw an exception if the interface does not exist. What do you think?

**Comment by oliverklee** [comment]
@sascha-egerer Like this? Or should we throw the exception in `src/Rule/RequestAttributeValidationRule.php`, too?

**Review by sascha-egerer — APPROVED** [review]

**Comment by sascha-egerer** [comment]
> @sascha-egerer Like this? Or should we throw the exception in `src/Rule/RequestAttributeValidationRule.php`, too?

For me it is fine like this.

**Pull request merged**